examples-gallery/two-pendulum-cart-variable-gallery.py

```diff
@@ -0,0 +1,311 @@
+lx, m1, m2, m3, g, iZZ1, iZZ2 = sm.symbols('lx, m1, m2, m3 g, iZZ1, iZZ2')             
+
+A1.orient_axis(N, q2, N.z)
+A1.set_ang_vel(N, u2 * N.z)
+A2.orient_axis(N, q3, N.z)
+A2.set_ang_vel(N, u3 * N.z)
+
+P1.set_pos(O, q1 * N.x)
+P2.set_pos(P1, lx * A1.x)
+P2.v2pt_theory(P1, N, A1)
+
+P3.set_pos(P2, lx * A2.x)
+P3.v2pt_theory(P2, N, A2)
+
+P1a = me.Particle('P1a', P1, m1)                                                
+
+I = me.inertia(A1, 0, 0, iZZ1)
+P2a = me.RigidBody('P2a', P2, A1, m2, (I, P2))                                 
+
+I = me.inertia(A2, 0, 0, iZZ2)
+P3a = me.RigidBody('P3a', P3, A2, m3, (I, P3))                                  
+BODY = [P1a, P2a, P3a]
+
+FL = [(P1, F * N.x - m1*g*N.y), (P2, -m2*g*N.y), (P3, -m3*g*N.y)]                   
+kd = sm.Matrix([q1.diff(t) - u1, q2.diff(t) - u2, q3.diff(t) - u3])                   
+
+q_ind = [q1, q2, q3]
+u_ind = [u1, u2, u3]
+
+KM = me.KanesMethod(N, q_ind=q_ind, u_ind=u_ind, kd_eqs=kd)
+(fr, frstar) = KM.kanes_equations(BODY, FL)
+EOM = kd.col_join(fr + frstar)
+sm.pprint(EOM)                
+
+# %%
+# Define various objects to be use in the optimization problem.
+h = sm.symbols('h')          # the variable time element
+
+state_symbols = tuple((*q_ind, *u_ind))
+constant_symbols = (lx, m1, m2, m3, g, iZZ1, iZZ2)
+specified_symbols = (F,)
+
+target_angle = np.pi /2.0                     
+num_nodes = 250     
+
+duration = (num_nodes - 1) * h
+interval_value = h
+
+# Specify the known system parameters.
+par_map = OrderedDict()
+par_map[lx]   = 2.0
+par_map[m1]  = 1.0
+par_map[m2]  = 1.0
+par_map[m3]  = 1.0
+par_map[g]   = 9.81
+par_map[iZZ1] = 2.0
+par_map[iZZ2] = 2.0
+
+def obj(free):
+    """Minimize h, the time interval between nodes."""
+    return free[-1]
+
+def obj_grad(free):
+    grad = np.zeros_like(free)
+    grad[-1] = 1.
+    return grad
+
+# %%
+# set the initial and final states to form the instance constraints.
+# initially, the pendulum should be hanging straight down, without mvoing
+# and finally, the pendulum should be upright, without moving.
+initial_state_constraints = {
+                            q1: 0.,  
+                            q2: -np.pi/2., 
+                            q3: -np.pi/2., 
+                            u1: 0.,
+                            u2: 0., 
+                            u3: 0.
+                            }
+
+final_state_constraints  = {
+                           q2: target_angle, 
+                           q3: target_angle, 
+                           u1: 0., 
+                           u2: 0., 
+                           u3: 0.
+                           }
+
+instance_constraints = (
+) + tuple(
+    xi.subs({t: 0}) - xi_val for xi, xi_val in initial_state_constraints.items()
+) + tuple(
+    xi.subs({t: duration}) - xi_val for xi, xi_val in final_state_constraints.items()
+)
+# bounding h > 0 herlps avoid 'solutions' with h < 0.
```

Comment:
It is best if the comments are made into the the sphinx-gallery rST sections so that they render as normal text in the website.

Reply:
See how the comments tied to `# %%` render nicely and the others do not:
![image](https://github.com/csu-hmc/opty/assets/276007/09fa178a-aa4c-4938-adc9-ea650d6d05bc)